Assets/Scripts/Combat/Abilities/Confection.cs

```diff
@@ -0,0 +1,277 @@
+        }
+        else
+        {
+            //end bake
+            EnableCanvas(BakeCanvas, false);
+            phase = Phase.Inactive;
+            CalculateBakeDamage();
+            EndAbility();
+        }
+    }
+
+    private void EndBrewPhase()
+    {
+        Debug.Log($"Brew Complete with candy: {SweetsDropped}");
+        Debug.Log($"Brew Complete with rots: {RotsDropped}");
+        //Reset sweets/rots
+        DragAndDrop[] allChildren = BrewCanvas.GetComponentsInChildren<DragAndDrop>();
+        foreach (DragAndDrop child in allChildren)
+        {
+            child.ResetPosition();
+        }
+
+        EnableCanvas(BrewCanvas, false);
+        CurrentDamage = (int) CalculateBrewDamage();
+        SweetsDropped = 0;
+        RotsDropped = 0;
+        CookingPot.ResetValues();
+        Debug.Log($"Brewing Build Up Damage: {CurrentDamage}");
+
+        StartBakePhase();
+    }
+
+    protected override void EndAbility()
+    {
+        //Canvas.ForceUpdateCanvases();
+        EnableCanvas(BrewCanvas, false);
+        EnableCanvas(BakeCanvas, false);
+        Debug.Log($"Confection Damage total: {CurrentDamage}");
+
+        //Only deals damage to one enemy
+        Attack attack = new Attack(CurrentDamage);
+        TargetedCombatants[Random.Range(0, TargetedCombatants.Length)].GetComponent<Combatant>().Defend(attack);
+        CombatSystem.EndTurn(this.GetComponentInParent<Combatant>().gameObject);
+    }
+
+    protected override void ContinueAbilityAfterTargeting()
+    {
+        StartBrewPhase();
+    }
+
+    private void StartBrewPhase()
+    {
+        phase = Phase.Brew;
+        SweetsDropped = 0;
+        RotsDropped = 0;
+
+        if(BrewCanvas != null)
+        {
+            //Move canvas to middle of the screen
+            Vector3 relativeScreenPosition = Camera.main.WorldToScreenPoint(transform.position);
+            BrewCanvas.transform.position = relativeScreenPosition;
+            EnableCanvas(BrewCanvas, true);
+        }
+        else
+        { 
+            Debug.Log("Canvas is Null in StartBrewPhase()"); 
+        }
+
+        timer.StartTimer(BrewDuration);
+    }
+
+    private void StartBakePhase()
+    {
+        phase = Phase.Bake;
+        NumberOfClicks = 0;
+
+        if(BakeCanvas != null)
+        {
+            //Move canvas to middle of the screen
+            Vector3 relativeScreenPosition = Camera.main.WorldToScreenPoint(transform.position);
+            BakeCanvas.transform.position = relativeScreenPosition;
+            EnableCanvas(BakeCanvas, true);
+            SliderScript.StartSlider();
+        }
+        else
+        { 
+            Debug.Log("Canvas is Null in StartBakePhase()"); 
+        }
+
+        timer.StartTimer(BakeDuration);
+    }
+
+    private float CalculateBrewDamage()
+    {
+        CurrentDamage = (SweetsDropped * SweetsDamage) + (RotsDropped * RottenDamage);
+        if(CurrentDamage <= 0)
+            CurrentDamage = BaseDamage;
```

Comment:
This is just a nitpick, but you could use `Mathf.Min()` instead

Reply:
I would remove that statement altogether. 

`CurrentDamage = (SweetsDropped * SweetsDamage) + (RotsDropped * RottenDamage) + BaseDamage;` would be better imo

Reply:
I will use Mathf.Max()
I can't use Daniel's because RottenDamage = -10 so there's a possibility that if only rotten items are dropped, the overall damage will be negative.